src/index.ts

```diff
@@ -2,12 +2,28 @@ import config from '@infrastructure/config/index.js';
 import logger from '@infrastructure/logging/index.js';
 import API from '@presentation/index.js';
 import runMetricsServer from '@infrastructure/metrics/index.js';
+import NoteService from '@domain/service/note.js';
+import { NoteRepository } from '@repository/index.js';
 
 const api = new API(config.httpApi);
 
 const start = async (): Promise<void> => {
   try {
-    await api.run();
+    /**
+     * TODO - Add database connection and creating storage instance
+     */
+
+    /**
+     * Initiate repositories
+     */
+    const noteRepository = new NoteRepository({});
+
+    /**
+     * Initiate services
+     */
+    const noteService = new NoteService(noteRepository);
```

Comment:
Let's move creation of services to `domain/index.ts`:

**index.ts**

```ts
import { init as initDomains } from '@/domain/index.ts'

```ts
const { noteService } = initDomains()
```

**domain/index.ts**

```ts
export function init(){
  const noteRepository = new NoteRepository({});
  const noteService = new NoteService(noteRepository);

  return {
    noteService
  }
}

```

Reply:
done